src/components/templates/Camp/hooks/usePartygoersOverlay.tsx

```diff
@@ -35,37 +32,33 @@ export const usePartygoersOverlay: ReactHook<
   partygoers,
   setSelectedUserProfile,
 }) => {
-  return useMemo(
-    () =>
-      // @debt this can be undefined because our types are broken so check explicitly
-      showGrid ? (
-        partygoers?.map(
-          (partygoer) =>
-            partygoer?.id && ( // @debt workaround, sometimes partygoers are duplicated but the new ones don't have id's
-              <MapPartygoerOverlay
-                key={partygoer.id}
-                partygoer={partygoer}
-                venueId={venueId}
-                myUserUid={userUid ?? ""} // @debt fix this to be less hacky
-                totalRows={rows}
-                totalColumns={columns}
-                withMiniAvatars={withMiniAvatars}
-                setSelectedUserProfile={setSelectedUserProfile}
-              />
-            )
-        )
-      ) : (
-        <div />
-      ),
-    [
-      showGrid,
-      partygoers,
-      venueId,
-      userUid,
-      rows,
-      columns,
-      withMiniAvatars,
-      setSelectedUserProfile,
-    ]
-  );
+  return useMemo(() => {
+    if (!showGrid) return null;
```

Comment:
[nit] new line after guard statements

```suggestion
    if (!showGrid) return null;
    
```